src/sys/windows/tcp.rs

```diff
@@ -412,9 +425,12 @@ impl Drop for TcpStream {
         // This is achieved by replacing our socket with an invalid one, so all
         // further operations will return an error (but no further operations
         // should be done anyway).
-        self.inner().socket = unsafe {
+        inner.socket = unsafe {
             net::TcpStream::from_raw_socket(INVALID_SOCKET)
         };
+
+        // Then run any finalization code including level notifications
+        inner.iocp.deregister2();
```

Comment:
Perhaps this could just be `let _ = inner.iocp.deregister()`? (instead of adding a new function)